### PR TITLE
[WIP] Varnish Vcl generator command

### PR DESCRIPTION
**Pull request by piotrkwiecinski**
Add new CLI command to allow generation of Varnish VCL configuration

### Description
In addition to new CLI to generate VCL file same generator is now used in admin UI.

### Fixed Issues (if relevant)
1. magento/magento2#9199: Create New CLI command: Generate Varnish VCL file

### Manual testing scenarios
1. bin/magento varnish:vcl:generate

Automated tests will be added tomorrow

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/PageCache/Console/Command/GenerateVclCommand.php [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Backend host for configuration',
+                'localhost'
+            ),
+            new InputOption(
+                self::BACKEND_PORT_OPTION,
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Backend post for configuration',
```
Comment (adragus-inviqa) [inline app/code/Magento/PageCache/Console/Command/GenerateVclCommand.php:184]: Typo: `port`, not `post`. Also, I had a bit of problem understanding the phrase `Backend port for configuration`. Can we be more explicit? I know what it does, but I just think some people will need more info about it. Don't be terse on documentation.

Comment (vrann) [reply app/code/Magento/PageCache/Console/Command/GenerateVclCommand.php:184]: Thanks for the comment. Do you have suggestions how to better call it?

**Comment by magento-team** [comment]
@piotrkwiecinski thank you for your contribution

**Comment by miguelbalparda** [comment]
@piotrkwiecinski I just saw this and I wanted to understand the code better. I couldn't find a switch for the admin value in the code and I wanted to know if this is not necessary or it is just not taken into account in this PR. Thanks!

**Comment by Ctucker9233** [comment]
@magento-team I see this PR isn't in 2.2 as of 2.2.7. Is there a backport or is this a 2.3 fix only?

**Pull request merged**